lib/gendered/guesser.rb

```diff
@@ -35,17 +36,41 @@ def guess!
 
           name
         end
+      else
+        message = sprintf "%s (%s)", guesses["message"], guesses["code"]
+        GenderedError.new(message)
       end
     end
 
+    # TODO: one can just call HTTP.get(url, :params => { ... })
     def url
       url = "https://api.genderize.io/?"
-      url += "country_id=#{country_id}&" if country_id
+      query = []
 
-      name_queries = names.collect.with_index do |name, index|
-        "name[#{index}]=#{CGI.escape(name.to_s)}"
+      [:country_id, :language_id, :apikey].each do |param|
+        next if @options[param].nil?
+        query << sprintf("%s=%s", param.to_s, CGI.escape(@options[param].to_s))
       end
-      url + name_queries.join("&")
+
+      names.each_with_index do |name, index|
+        query << sprintf("name[%s]=%s", index, CGI.escape(name.to_s))
```

Comment:
Specifying the index should not be needed, but I kept it anyways, maybe I'm missing something...